lib/livestatus/livestatuslogutility.cpp

```diff
@@ -80,11 +80,24 @@ void LivestatusLogUtility::CreateLogCache(std::map<time_t, String> index, Histor
 
 	/* m_LogFileIndex map tells which log files are involved ordered by their start timestamp */
 	unsigned long line_count = 0;
-	for (const auto& kv : index) {
+	std::map<time_t, String>::const_iterator last = index.end();
+	--last;
+
+	for ( std::map<time_t, String>::const_iterator iter = index.begin() ; iter !=index.end() ; ++iter) {
+		const std::pair<time_t, String>& kv = *iter ;
 		unsigned int ts = kv.first;
+		unsigned int ets;
+
+		/* using previous next start as end of this log or now if last entry, aka current log*/
+		if ( iter == last ) {
+			ets=time(NULL);
```

Comment:
time(NULL) equals Utility::GetTime(), the integer part of it.

Reply:
I've arranged like this to follow implicit ts cast from time_t i'll test with GetTime